### ISSUE TITLE
Query not returning the desired number of rows all the time

### ISSUE THREAD

**Issue by alexch-rawa** [arangodb/arangodb#13303]
## My Environment

* __ArangoDB Version__:        3.5.2 upgraded to 3.7.3
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Single Server
* __Deployment Strategy__:   Manual Start in Docker
* __Configuration__:               nothing special
* __Infrastructure__:               deployed on a EC2
* __Operating System__:        Ubuntu 16.04
* __Total RAM in your machine__:        16GB
* __Disks in use__:        GP2
* __Used Package__:              Docker - official Docker library

## Component, Query & Data

__AQL query (if applicable)__:

(please check __Problem__ and __Expected Result__)

```
FOR user IN user
    FILTER user.mySqlId IN [4084,4336,11024,11811,15328]

    LET receiveGifts = (
        FOR ua IN userAbility
            FILTER ua._from == user._id
            AND ua._to == (FOR a IN ability FILTER a.name == 'RECEIVE_GIFTS' AND a.isActive == true RETURN a._id)[0]
            AND ua.isActive == true
            AND ua.isComplete == true
        RETURN ua.isComplete
    )
    
    RETURN {
        mySqlId: user.mySqlId,
        arangoId: user._id,
        xp: user.xp,
        level: user.level,
        canReceiveGifts: receiveGifts[0],
        coins: user.coins
}
```

__Dataset__:

Copy/pasting as text, sorry:

Users:
```
_id:user/71895
_rev:_aDymKQO--I
_key:71895

{
  "mySqlId": 4084,
  "level": 1,
  "xp": 0,
  "coins": 0,
  "createdAt": 1580671739,
  "updatedAt": 1582110862989,
  "deletedAt": null,
  "isActive": true
}
```

```
_id:user/169514
_rev:_bofZkem---
_key:169514

{
  "mySqlId": 4336,
  "level": 74,
  "xp": 8691,
  "coins": 653095,
  "createdAt": 1582148152335,
  "updatedAt": 1609142344202,
  "deletedAt": null,
  "isActive": true
}
```

```
_id:user/10130719
_rev:_bogAmja---
_key:10130719

{
  "mySqlId": 11024,
  "level": 34,
  "xp": 1042,
  "coins": 5,
  "createdAt": 1592950221981,
  "updatedAt": 1609144902230,
  "deletedAt": null,
  "isActive": true
}
```

```
_id:user/10814321
_rev:_bok4Fau---
_key:10814321

{
  "mySqlId": 11811,
  "level": 47,
  "xp": 1303,
  "coins": 43505,
  "createdAt": 1593544052737,
  "updatedAt": 1609165315532,
  "deletedAt": null,
  "isActive": true
}
```

```
_id:user/17297646
_rev:_bom-J1----
_key:17297646

{
  "mySqlId": 15328,
  "level": 44,
  "xp": 386,
  "coins": 2415,
  "createdAt": 1597952418686,
  "updatedAt": 1609169907568,
  "deletedAt": null,
  "isActive": true
}
```

Ability:
```
_id:ability/72422
_rev:_aDymTBW--C
_key:72422

{
  "createdAt": 1582110888977,
  "updatedAt": 1582110888977,
  "deletedAt": null,
  "isActive": true,
  "title": {
    "en": "Receive Gifts"
  },
  "name": "RECEIVE_GIFTS",
  "type": "MY-SQL-ATTAINABLE",
  "data": {
    "type": "feature",
    "mySqlAttainableId": 7,
    "mySqlChannelId": 1
  }
}
```

UserAbility:
```
_id:userAbility/2555413
_rev:_aKYCu56---
_key:2555413
_from:user/169514
_to:ability/72422

{
  "createdAt": 1583878554559,
  "updatedAt": 1583878554559,
  "deletedAt": null,
  "isActive": true,
  "isComplete": true,
  "acquiredAt": 1583878554559,
  "completedAt": 1583878554559
}
```

```
_id:userAbility/13882537
_rev:_a1i1Mn----
_key:13882537
_from:user/10814321
_to:ability/72422

{
  "createdAt": 1595466529424,
  "updatedAt": 1595466529424,
  "deletedAt": null,
  "isActive": true,
  "isComplete": true,
  "acquiredAt": 1595466529424,
  "completedAt": 1595466529424
}
```

__Size of your Dataset on disk__:

308MB

## Steps to reproduce

1. Run the query, with `IN [4084,4336,11024,11811,15328]`
2. Run the query, with `IN [4084,11024,11811,15328]`
3. Run the query, with `IN [11024,11811,15328]`

__Problem__:

Running step 1, I get 5 rows returned:

```
[
  {
    "mySqlId": 4084,
    "arangoId": "user/71895",
    "xp": 0,
    "level": 1,
    "canReceiveGifts": null,
    "coins": 0
  },
  {
    "mySqlId": 4336,
    "arangoId": "user/169514",
    "xp": 10919,
    "level": 73,
    "canReceiveGifts": true,
    "coins": 4408015
  },
  {
    "mySqlId": 11024,
    "arangoId": "user/10130719",
    "xp": 1276,
    "level": 33,
    "canReceiveGifts": null,
    "coins": 0
  },
  {
    "mySqlId": 11811,
    "arangoId": "user/10814321",
    "xp": 590,
    "level": 46,
    "canReceiveGifts": true,
    "coins": 35265
  },
  {
    "mySqlId": 15328,
    "arangoId": "user/17297646",
    "xp": 4103,
    "level": 41,
    "canReceiveGifts": null,
    "coins": 5095
  }
]
```

Running step 2, I get 2 rows returned:

```
[
  {
    "mySqlId": 4084,
    "arangoId": "user/71895",
    "xp": 0,
    "level": 1,
    "canReceiveGifts": null,
    "coins": 0
  },
  {
    "mySqlId": 11024,
    "arangoId": "user/10130719",
    "xp": 1276,
    "level": 33,
    "canReceiveGifts": null,
    "coins": 0
  }
]
```

Running step 3, I get 1 row returned:
```
[
  {
    "mySqlId": 11024,
    "arangoId": "user/10130719",
    "xp": 1276,
    "level": 33,
    "canReceiveGifts": null,
    "coins": 0
  }
]
```

__Expected result__:

I get one row returned for each ID in the FILTER.

If this is a valid issue, and you need further clarifications/logs/dump, I can make them available privately.

**Comment by dothebart** [comment]
Hi, 
can you please check whether the issue persists with the recently published 3.7.5?

**Comment by alexch-rawa** [comment]
> Hi,
> can you please check whether the issue persists with the recently published 3.7.5?

Checked, same behaviour.

**Comment by alexch-rawa** [comment]
Duplicate of #13305
